plugins/woocommerce/src/Internal/DataStores/StockNotifications/StockNotificationsDataStore.php

```diff
@@ -0,0 +1,384 @@
+<?php // phpcs:ignore Suin.Classes.PSR4
+/**
+ * StockNotificationsDataStore class file.
+ */
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\DataStores\StockNotifications;
+
+use Automattic\WooCommerce\Internal\Utilities\DatabaseUtil;
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * The Stock Notifications Data Store.
+ */
+class StockNotificationsDataStore implements \WC_Object_Data_Store_Interface {
```

Comment:
I thought about the data store(s) a bit more and thought we might be able to get the best of both worlds:
- nice abstraction and encapsulation, and
- separation of concerns  
 
...if we used an umbrella data store as an abstraction for the stock notifications and dependency-inject 2 sub-data store classes for meta and log. Wdyt?

Reply:
Completely agree; the primary data store is becoming unnecessarily large.